dotnet/samples/04.e.twentyQuestions/ResponseBuilder.cs

```diff
@@ -141,10 +139,7 @@ public static string QuitGame(string? secretWord)
 
         public static string YouLose(string secretWord) => string.Format(GetRandomResponse(LOSE_MESSAGES), secretWord);
 
-        private static string GetRandomResponse(string[] responses)
-        {
-            int index = RANDOM.Next(responses.Length);
-            return responses[index];
-        }
+        private static string GetRandomResponse(string[] responses) =>
+            responses[Random.Shared.Next(responses.Length)];
```

Comment:
I'm not a `dotnet` dev; could you explain this one to me? :)

Reply:
The previous version was using a `private static Random RANDOM = new Random()` instance. I don't know what the threading model is of these teams bots, but if there's any chance that multiple threads could concurrently end up calling this GetRandomResponse method, that would not be safe: like most .NET types, Random is not safe to be used by multiple threads concurrently and risks being corrupted with such use. Instead, the change switches to using the instance returned from `Random.Shared`, which _is_ thread-safe, so this is just switching from using a non-thread-safe instance to a thread-safe instance.  This line is saying "give me a random value in the range [0, response.Length), and return the value from responses at that index".